File: day02/sol.lua
print('Day 2: Lua')
local p1, p2 = 0, 0
for line in io.stdin:lines() do
   local a = line:byte(1) - ('A'):byte()
   local b = line:byte(3) - ('X'):byte()
   local wld = b == (a+1)%3 and 6 or b == a and 3 or 0
   p1 = p1 + wld + b + 1

   b = (a+b-1)%3
   wld = b == (a+1)%3 and 6 or b == a and 3 or 0
   p2 = p2 + wld + b + 1
end
print(string.format('Part 1: %20d', p1))
print(string.format('Part 2: %20d', p2))

File: day25/sol.nim
import algorithm
import math
import strformat
import strutils

echo "Day 25: Nim"
var num = 0
var line: string
while stdin.readLine(line):
  for i, c in line.reversed:
    num += 5^i * ("=-012".find(c) - 2)
var p1: string
while num > 0:
  p1.add "012=-"[num mod 5]
  num = (num + 2) div 5
p1.reverse
echo fmt"Part 1: {p1:20}"
echo "Part 2: $1".format(align("n/a", 20))
